add query console/code search info (#1528)
semgrep/semgrep-docs: docs/semgrep-code/editor.md

<EnableTurboMode />

## Code search (beta)

> Code search allows you to test a Semgrep rule by running it against one or more GitHub repositories instead of just a few lines of test code. Its results highlight all instances of matching code in your repositories, allowing you to see whether your rule works as intended or not. This rapid feedback can help you develop more accurate and effective rules.

Reviewer: ```suggestion
Code search allows you to test a Semgrep rule by running it against one or more GitHub repositories or projects instead of just a few lines of test code. Its results highlight all instances of matching code in those target repositories, allowing you to see whether your rule works as intended or not. This rapid feedback can help you develop more accurate and effective rules.
```

![Code search in Semgrep Editor](/img/code-search.png)
***Figure.*** Code search in Semgrep Editor

> :::info prerequisites

Reviewer: Convert to heading?

> :::info prerequisites

Author: Just Github.com right now

> * Code search is currently available to all paying customers of Semgrep Code.

Reviewer: Checking this wording with the pm-pmm team!

> * You must have granted Semgrep code access if you would like to run code search against your repositories. Otherwise, you can run code search against public repositories.

Reviewer: ```suggestion
* You must have granted Semgrep code access if you would like to run code search against your repositories. Otherwise, you can run code search against public repositories.
```
- Can we change this to present tense? Also, does this need Semgrep Cloud Scanning or just "code access"?

Reviewer: - Just "Code access" - can be granted by enabling Assistant as it creates a private Semgrep app, doesn't require ZCS, works on CI-based scans
- Needs Cloud/Managed scanning - needs more than just a private app, does not work on CI-based scans.

:::

> To run your rule against your repositories:

Reviewer: ```suggestion
To run your rule against selected repositories or projects:
```

1. Sign in to [Semgrep AppSec Platform](https://semgrep.dev/login).
2. Go to **Rules > Editor**, and open up the rule you want to test.

> 3. In the **test code** panel, switch to **live code**.

Reviewer: ```suggestion
3. In the **code panel** click **live code**.
```

> 4. Select the repositories against which you want the rule to run. You can use the search bar to narrow down the list of repositories shown. Semgrep currently supports both public and private repositories. 

Reviewer: ```suggestion
4. Select the repositories against which you want the rule to run. You can use the search bar to narrow down the list of repositories shown. Semgrep currently supports both public and private repositories. 
```
There's a caveat with these private repositories, right?

<EnableTurboMode />

## Code search (beta)

Code search allows you to test a Semgrep rule by running it against one or more GitHub repositories or projects instead of just a few lines of test code. Its results highlight all instances of matching code in those target repositories, allowing you to see whether your rule works as intended or not. This rapid feedback can help you develop more accurate and effective rules.

![Code search in Semgrep Editor](/img/code-search.png)
***Figure.*** Code search in Semgrep Editor

### Prerequisites

> * Code search is currently available to all paying customers of Semgrep Code.

Reviewer: ```suggestion
* Code search is in **public beta** for Semgrep customers on a paid plan.
```

https://semgrepinc.slack.com/archives/C05H6SFG1DL/p1715355000385759

> * You must grant Semgrep code access if you would like to run code search against your repositories. Otherwise, you can run code search against public repositories. 

Reviewer 2: I think this needs a rephrasing to remove "code access" -- the way to access is by going through the "Register your Github App" flow!

Reviewer 2: The rest of this looks great though!

Reviewer: So both ZCS and query console require the "Register your GitHub App" flow - I have an existing [set of instructions here](https://deploy-preview-1536--semgrep-docs-prod.netlify.app/docs/deployment/managed-scanning#install-the-private-semgrep-github-app) that have been approved by Jack & Pablo. I wonder if we want to modularize it so we can reuse it in this doc, or just link to it when it is live?
